streams/src/commonMain/kotlin/com/mirego/trikot/streams/reactive/PublisherExtensions.kt

```diff
@@ -252,3 +254,19 @@ fun <T> Publisher<T>.retryBackoff(
         }
     }
 }
+
+/**
+ * The TakeUntil uses a predicate function that evaluates the items emitted by the source Publisher
+ * to terminate if we mirrors the source OR emit one last item and complete immediately after.
+ *
+ * Marbles diagram :
+ * -------(1)---(2)-----(3)-----(4)--|->
+ * takeUntil(==3)
+ * -------(1)---(2)-----(3)/----------->
+ *
+ * @see @see <a href="http://reactivex.io/documentation/operators/takeuntil.html">http://reactivex.io/documentation/operators/takeuntil.html</a>
```

Comment:
double see

Reply:
The linked documentation is not representative of this implementation. The one linked mentions : 

> discard any items emitted by an Observable after a second Observable emits an item or terminates

We should find another link to this **predicate** version of the operator.

Reply:
You have to scroll down and open the tab "RxJava 1․x takeUntil" to see both versions. Or I can put the link to RxJava directly: http://reactivex.io/RxJava/javadoc/rx/Observable.html#takeUntil-rx.functions.Func1-